src/wallet/wallet.cpp

```diff
@@ -1889,6 +1889,16 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived,
         }
     }
 
+    // Create an output for value taken from (negative valueBalance) or added (positive valueBalance)
+    // to the transparent value pool by Sapling shielding and unshielding.
+    if (valueBalance < 0) {
+        COutputEntry output = {CNoDestination(), -valueBalance, (int)vout.size()};
+        listSent.push_back(output);
+    } else if (valueBalance > 0) {
+        COutputEntry output = {CNoDestination(), valueBalance, (int)vout.size()};
+        listReceived.push_back(output);
+    }
```

Comment:
Should this be inside the `if (isFromMyTaddr) {` scope, rather than outside? Needs checking.

Reply:
I agree it should be inside, like the two blocks above it.  Pushed commit for this.